### PR TITLE
52 and 54 providers new/create and read/show

### PR DESCRIPTION
**Pull request by JorjaF**
### Description of changes
I added the provider model, controller, table, and API call for new/create provider
Added show function to provider and did sad path

### Test Suite
- [x] Are all tests within the test suite passing?

### Specific Feedback Request(s)
anything you got

#### Add GIF (NOT Optional)
image

### CODE REVIEW

File: spec/requests/api/v0/provider_spec.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+          'contact_phone' => '123-456-7890',
+          'fees' => 'Test Fees',
+          'schedule' => 'Test Schedule'
+        )
+      end
+    end
+
+    context "when the provider does not exist" do
+      it "returns a 404 (Not Found) status" do
+        get "/api/v0/providers/9999"
```
Comment (westonio) [inline spec/requests/api/v0/provider_spec.rb:98]: Just for the sake of if we end up using factorybot, I would recommend using a much larger number or even do something like:

`id = Float::INFINITY` <- this is the largest number possible in ruby, always
`get "/api/v0/providers/#{id}"`

**Review by westonio — CHANGES_REQUESTED** [review]
Hi @JorjaF ! Everything looks amazing overall 😄  I'm just adding a request for some minor changes and recommendation if you see fit

File: app/controllers/api/v0/providers_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+module Api
+  module V0
+    class ProvidersController < ApplicationController
```
[the remark this answers is not in the document]
Comment (JorjaF) [reply app/controllers/api/v0/providers_controller.rb:3]: I fixed this

File: spec/requests/api/v0/provider_spec.rb [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+          city: "Test City",
+          state: "TX",
+          zip: "12345",
+          web_url: "https://www.test.com",
+          contact_phone: "123-456-7890",
+          fees: "Test Fees",
+          schedule: "Test Schedule"
+        }
+      end
+
```
[the remark this answers is not in the document]
Comment (JorjaF) [reply spec/requests/api/v0/provider_spec.rb:20]: I updated this

File: spec/requests/api/v0/provider_spec.rb [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      it "does not create a provider" do
+        expect do
+          post "/api/v0/providers", params: { provider: invalid_params }
+        end.to change(Provider, :count).by(0)
+
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
```
[the remark this answers is not in the document]
Comment (JorjaF) [reply spec/requests/api/v0/provider_spec.rb:52]: I added the test for that

File: spec/requests/api/v0/provider_spec.rb (hunk 1 above)
Comment (JorjaF) [reply spec/requests/api/v0/provider_spec.rb:98]: Updated this too

**Comment by z-fitch** [comment]
Thanks for making those changes! It looks good now!

**Review by westonio — APPROVED** [review]
Thank you @JorjaF for making all those updates! Looks great 🙌

**Pull request merged**